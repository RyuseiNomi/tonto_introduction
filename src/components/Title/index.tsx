import React from 'react';
import './index.css';

type Props = {
  hook: Function;
  toggle: Function;
  scrollDirection: string;
  language: string;
}

function Title(props: Props) {

  const { hook, toggle, scrollDirection, language } = props;

  // 言語切替ボタンで enabled な状態
  const languageSelectedStyle = {
    backgroundColor: '#E8C1B8',
    color: 'white',
    borderColor: 'white'
  }

  // 言語切替ボタンで disabled な状態
  const languageNotSelectedStyle = {
    backgroundColor: 'white',
    color: '#E89A8A',
    borderColor: 'white'
  };
  return (
    <div className="title-bar">
      <div className="language-selection">
        <button
          type="button"
          onClick={() => hook("jp")}
          style={language === "jp" ? languageSelectedStyle : languageNotSelectedStyle}
          className="language-selection-button"
        >
          JP
        </button>
        <button
          type="button"
          onClick={() => hook("en")}
          style={language === "en" ? languageSelectedStyle : languageNotSelectedStyle}
          className="language-selection-button"
        >
          EN
        </button>
      </div>
      <picture data-testid="title-image">
      {scrollDirection === "down" ?
        <source
          type="image/webp"
          srcSet={
            language === "jp" ? `${process.env.PUBLIC_URL}/assets/title_jp.webp` : `${process.env.PUBLIC_URL}/assets/title_en.webp`
          }
        >
        </source>
        :
        <source
          type="image/webp"
          srcSet={`${process.env.PUBLIC_URL}/assets/title_hutaba.webp`}
        >
        </source>
      }
        {scrollDirection === "down" ?
          <img
            className="title-image"
            src={
              language === "jp" ? `${process.env.PUBLIC_URL}/assets/title_jp.png` : `${process.env.PUBLIC_URL}/assets/title_en.png`
            }
            alt="background"
          />
          :
          <img
            className="title-image"
            src={`${process.env.PUBLIC_URL}/assets/title_hutaba.png`}
            alt="background"
          />
        }
      </picture>
      <img
        className="background-image"
        src={`${process.env.PUBLIC_URL}/hutaba_background.jpg`}
        alt="background"
        onClick={() => toggle(true)}
      />
      {/* スクロールを促すアニメーション */}
      {scrollDirection === "down" ?
        <p className="scroll-down" data-testid="scroll"><span></span></p>
        :
        <p className="scroll-up" data-testid="scroll"><span></span></p>
      }
    </div>
  );
}

export default Title;
