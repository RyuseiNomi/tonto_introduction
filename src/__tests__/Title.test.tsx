import { cleanup, act } from '@testing-library/react';
import { screen } from '@testing-library/dom'
import { createRoot } from 'react-dom/client';
import renderer from 'react-test-renderer';
import Title from './../components/Title';

let container = null;
let root = null;
let prop_language:string = "";

// 言語切替用のダミー関数
const updateLanguageSetting = (language: string) => {
  prop_language = language;
};

beforeEach(() => {
  // conteiner の定義
  container = document.createElement("div");
  document.body.appendChild(container);
  root = createRoot(container);
});

afterEach(cleanup);

// =============== snapshot test =================
it('タイトル画面が表示されていること', () => {
  const component = renderer.create(<Title hook={updateLanguageSetting("jp")} language={prop_language} />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// ================ act tests ====================
it('日本語を選択している時、日本語用タイトル画像が表示されていること', async () => {
  act(() => {
    root.render(<Title hook={updateLanguageSetting("jp")} language={prop_language} />);
  })

  const webpImg = container.querySelector("source")
  const pngImg = container.querySelector("img")

  // img タグの存在検証
  expect(pngImg).toHaveAttribute("src", "/assets/title_jp.png");
  // source タグの存在検証
  expect(webpImg).toHaveAttribute("srcSet", "/assets/title_jp.webp");
});

it('英語を選択している時、英語用タイトル画像が表示されていること', async () => {
  act(() => {
    root.render(<Title hook={updateLanguageSetting("en")} language={prop_language} />);
  })

  const webpImg = container.querySelector("source")
  const pngImg = container.querySelector("img")

  // img タグの存在検証
  expect(pngImg).toHaveAttribute("src", "/assets/title_en.png");
  // source タグの存在検証
  expect(webpImg).toHaveAttribute("srcSet", "/assets/title_en.webp");
});
