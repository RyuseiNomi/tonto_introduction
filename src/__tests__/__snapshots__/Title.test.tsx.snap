// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`タイトル画面が表示されていること 1`] = `
<div
  className="title-bar"
>
  <div
    className="language-selection"
  >
    <button
      className="language-selection-button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E8C1B8",
          "borderColor": "white",
          "color": "white",
        }
      }
      type="button"
    >
      JP
    </button>
    <button
      className="language-selection-button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "white",
          "color": "#E89A8A",
        }
      }
      type="button"
    >
      EN
    </button>
  </div>
  <picture
    data-testid="title-image"
  >
    <source
      srcSet="/assets/title_jp.webp"
      type="image/webp"
    />
    <img
      alt="background"
      className="title-image"
      src="/assets/title_jp.png"
    />
  </picture>
  <img
    alt="background"
    className="background-image"
    src="/assets/kanon_tonto_white.png"
  />
  <p
    className="scroll-down"
    data-testid="scroll"
  >
    <span />
  </p>
</div>
`;
